### ISSUE TITLE
Incorrect unescaping of Docker output by escaped decoder

### ISSUE THREAD

**Issue by zaa** [fluent/fluent-bit#907]
## Bug Report

**Describe the bug**

I've executed the following commands in a docker container:

```
# docker run -it alpine:3.8
/ # echo тест
тест
/ # echo \\d\\e\\v > 1
/ # cat 1
\d\e\v
/ # exit
```

Docker generated the following entries in the json log file:

```
{"log":"/ # \u001b[6n\r/ # \u001b[Jecho тест\r\n","stream":"stdout","time":"2018-11-13T19:20:12.468784227Z"}
{"log":"тест\r\n","stream":"stdout","time":"2018-11-13T19:20:12.469447631Z"}
{"log":"/ # \u001b[6necho \\\\d\\\\e\\\\v \u003e 1\r\n","stream":"stdout","time":"2018-11-13T19:20:18.591492065Z"}
{"log":"/ # \u001b[6ncat 1\r\n","stream":"stdout","time":"2018-11-13T19:20:20.089090452Z"}
{"log":"\\d\\e\\v\r\n","stream":"stdout","time":"2018-11-13T19:20:20.139132068Z"}
{"log":"/ # \u001b[6nexit\r\n","stream":"stdout","time":"2018-11-13T19:20:22.611505788Z"}
```

I've processed the file using the following parser:

```
[PARSER]
  Name         docker
  Format       json
  Time_Key     time
  Time_Format  %Y-%m-%dT%H:%M:%S.%L
  Time_Keep    On
  Decode_Field_As escaped log
```

and saw the following output from the file plugin:

```
containers: [1542136812.468784, {"log":"/ # u001b[6n\r/ # u001b[Jecho \u0442\u0435\u0441\u0442\r\n", "stream":"stdout", "time":"2018-11-13T19:20:12.468784227Z"}]
containers: [1542136812.469448, {"log":"\u0442\u0435\u0441\u0442\r\n", "stream":"stdout", "time":"2018-11-13T19:20:12.469447631Z"}]
containers: [1542136818.591492, {"log":"/ # u001b[6necho \\\\d\\\\e\\\\v u003e 1\r\n", "stream":"stdout", "time":"2018-11-13T19:20:18.591492065Z"}]
containers: [1542136820.089090, {"log":"/ # u001b[6ncat 1\r\n", "stream":"stdout", "time":"2018-11-13T19:20:20.089090452Z"}]
containers: [1542136820.139132, {"log":"\\d\\e\\v\r\n", "stream":"stdout", "time":"2018-11-13T19:20:20.139132068Z"}]
containers: [1542136822.611506, {"log":"/ # u001b[6nexit\r\n", "stream":"stdout", "time":"2018-11-13T19:20:22.611505788Z"}]
```

As you can see, unicode escapes like `\u001b` and `\u003e` from the docker log turned in the resulting output into plain text like `u001b` and `u003e`.

As far as I understand, this happens because flb_unescape_string strips `\` from the input string if the symbol is not followed by a known escape literal like `n`, `r`, `b`, etc: https://github.com/fluent/fluent-bit/blob/master/src/flb_unescape.c#L219

I tried to replace `escaped` in the parser config with `escaped_utf8`, bit this produced even more incorrect output:

```
containers: [1542137905.038102, {"log":"/ # \u001b[6n\r/ # \u001b[Jecho \r\\n", "stream":"stdout", "time":"2018-11-13T19:38:25.038102479Z"}]
containers: [1542137905.038636, {"log":"\r\\n", "stream":"stdout", "time":"2018-11-13T19:38:25.038636564Z"}]
containers: [1542137913.983507, {"log":"/ # \u001b[6necho \\\\\\d\\\\\\e\\\\\\\v > 1\r\\n", "stream":"stdout", "time":"2018-11-13T19:38:33.983507342Z"}]
containers: [1542137918.904600, {"log":"/ # \u001b[6ncat 1\r\\n", "stream":"stdout", "time":"2018-11-13T19:38:38.904600072Z"}]
containers: [1542137918.904982, {"log":"\\d\\e\\\v\r\\n", "stream":"stdout", "time":"2018-11-13T19:38:38.904982423Z"}]
containers: [1542137920.275188, {"log":"/ # \u001b[6nexit\r\\n", "stream":"stdout", "time":"2018-11-13T19:38:40.275188481Z"}]
```

1. there is no "тест" output on the first and second lines
2. `echo \\\\d\\\\e\\\\v \u003e 1` from docker turned into `echo \\\\\\d\\\\\\e\\\\\\\v > 1` (more escaping than needed)
3. `\\d\\e\\\v\r\\n` - 3 slashes instead of 2 in front of `v`, 2 slashes in front of `n`

I also tried to remove the decoding altogether, but this produced incorrect output as well:

```
containers: [1542138466.634099, {"log":"\\\\d\\\\e\\\\v\\r\\n", "stream":"stdout", "time":"2018-11-13T19:47:46.634099441Z"}]
```

So I am a bit at loss which method to use. None of them are correct:
- `escaped` mangles `\uXXXX` escapes
- `escaped_utf8` mangles even more data
- absence of the decoder leads to excessive escaping

**Expected behavior**

Fluent-Bit preserves output from docker.

**Your Environment**
* Version used: Fluent-Bit v0.14.7
* Docker: 18.06.1-ce

**Comment by zaa** [comment]
Bump

**Comment by poinntt** [comment]
+1

**Comment by edsiper** [comment]
Please check the following comment on #1278 : 

https://github.com/fluent/fluent-bit/issues/1278#issuecomment-499583503

**Comment by edsiper** [comment]
Issue already fixed, ref: #1278 (comment)
